functions/Export-DbaRegServer.ps1

```diff
@@ -73,47 +82,96 @@ function Export-DbaRegServer {
         [object[]]$InputObject,
         [string]$Path = (Get-DbatoolsConfigValue -FullName 'Path.DbatoolsExport'),
         [Alias("OutFile", "FileName")]
-        [string]$FilePath,
+        [ValidateScript( {
+                if ($_.FullName -notmatch "\.xml$|\.regsrvr$") {
+                    throw "The FilePath specified must end with either .xml or .regsrvr"
+                }
+                return $true
+            })]
+        [System.IO.FileInfo]$FilePath,
         [ValidateSet("None", "PersistLoginName", "PersistLoginNameAndPassword")]
         [string]$CredentialPersistenceType = "None",
+        [object[]]$Group,
+        [object[]]$ExcludeGroup,
+        [switch]$Overwrite,
         [switch]$EnableException
     )
     begin {
         $null = Test-ExportDirectory -Path $Path
-        $timeNow = (Get-Date -uformat "%m%d%Y%H%M%S")
+        $timeNow = (Get-Date -UFormat "%m%d%Y%H%M%S")
+
+        # ValidateScript in the above param block relies on the order of the params specified by the user,
+        # so the creation of the file path and $Overwrite are evaluated here
+        if ($PSBoundParameters.ContainsKey("FilePath")) {
+            if (-not (Test-Path $FilePath) ) {
+                New-Item -Path $FilePath.DirectoryName -ItemType "directory" -Force | Out-Null # make sure the parent dir exists
+            } elseif (-not $Overwrite.IsPresent) {
+                throw [System.Management.Automation.RuntimeException] "Use the -Overwrite parameter if the file $FilePath should be overwritten."
```

Comment:
We use an internal command, `Stop-Function`, in place of throws. `Stop-Function` is user friendly in appearance and also does additional logging behind the scenes.

```suggestion
                Stop-Function -Message "Use the -Overwrite parameter if the file $FilePath should be overwritten."
                return
```